readme.md

```diff
@@ -65,61 +65,91 @@ go run ./examples/main.go
 
 ## Installing
 
+On your server:
+
 ```
 go get github.com/canopyclimate/golive@latest
 ```
 
-## About `htmltmpl`
+In your frontend:
 
-This package includes a fork of the standard library’s `html/template` package named `htmltmpl`. It has similar security guarantees to the original package. The fork exists to support the diffing needed to track changes in your HTML based on the live view’s state.
+```
+npm i --save phoenix
+npm i --save phoenx_live_view
+```
 
-## Routing
+And then add the following to whatever JavaScript you’ll be loading in your frontend (you probably want to load this file in whatever template your `live.Config.WriteLayout` writes out):
 
-As noted above, GoLive does not endeavor to own your app’s routing. Instead, we provide a few different ways to connect a given route in your app to a live view.
+```js
+import { Socket } from "phoenix";
+import { LiveSocket } from "phoenix_live_view";
 
-First, you always need to route the path your live views will use to mount themselves via websockets. You can do this by creating a `live.Config` and passing it into `live.NewWebsocketHandler`, giving you an `http.Handler` you can use as you see fit:
+let csrfToken = document.querySelector("meta[name='csrf-token']")?.getAttribute("content")
+// "/live" below should be whatever path you set up for your websocket.
+let liveSocket = new LiveSocket("/live", Socket, {params: {_csrf_token: csrfToken}})
 
-```
-r.Handle("/live/websocket", live.NewWebsocketHandler(liveConfig))
+// connect if there are any LiveViews on the page
+liveSocket.connect()
 ```
 
-Also in this `live.Config` is a `Mux` property that accepts any `http.Handler` that can mux to your live views. Actually creating the view in your handlers can happen in a few different ways. Each of the options below assume you’re in the context of some handler that has an `http.ResponseWriter` named `w`:
+## About `htmltmpl`
 
-First, you can explicitly cast your response writer to one that understands live views and, if successful, return the view:
+This package includes a fork of the standard library’s `html/template` package named `htmltmpl`. It has similar security guarantees to the original package. The fork exists to support the diffing needed to track changes in your HTML based on the live view’s state.
 
-```go
-if j, ok := w.(*live.JoinHandler); ok {
-    j.SetView(new(Counter))
-}
-```
+## Routing & Config
+
+As noted above, GoLive does not endeavor to own your app’s routing. Instead, we provide a `net/http` middleware which you then use—any compatible routing library will do.
```

Comment:
We should say something about why it is middleware.  Something a long lines of:  the middleware checks first to see whether the HTTP handler set an associated live view. If not, it falls back to your regular http handlers.